python/matcher.go

```diff
@@ -19,40 +21,65 @@ type Matcher struct{}
 // Name implements driver.Matcher.
 func (*Matcher) Name() string { return "python" }
 
-// Filter implements driver.Matcher.
-func (*Matcher) Filter(record *claircore.IndexRecord) bool {
-	return record.Package.NormalizedVersion.Kind == "pep440"
+func (*Matcher) Filter(r *claircore.IndexRecord) bool {
+	return r.Package.NormalizedVersion.Kind == "pep440"
 }
 
 // Query implements driver.Matcher.
 func (*Matcher) Query() []driver.MatchConstraint {
-	return []driver.MatchConstraint{}
+	return []driver.MatchConstraint{driver.RepositoryName}
 }
 
 // Vulnerable implements driver.Matcher.
 func (*Matcher) Vulnerable(ctx context.Context, record *claircore.IndexRecord, vuln *claircore.Vulnerability) (bool, error) {
-	// if the vuln is not associated with any package,
-	// return not vulnerable.
-	if vuln.Package == nil {
-		return false, nil
+	if vuln.FixedInVersion == "" {
+		return true, nil
+	}
+
+	decodedVersions, err := url.ParseQuery(vuln.FixedInVersion)
+	if err != nil {
+		return false, err
+	}
+	upperVersion := decodedVersions.Get("fixed")
+	if upperVersion == "" {
+		return false, fmt.Errorf("osv: missing upper version")
```

Comment:
thinking about it, maybe this should say `python` instead of `osv` because, even though the data is OSV, the function is under the `python` directory. I'll let someone else chime in about this because I'm not 100% sure the convention used in this repo

Reply:
agree